Make sure the updated date is still output in the HTML

laurelfulford opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

When the option was added to 'Show updated date' on single posts in #1154, the theme's original behaviour -- always outputting and hiding the updated date -- stopped working. 

This is because the check for whether the published date matched the modified date got moved into the `newspack_should_display_updated_date()` check, so it was only run when the post would display the updated date when `Show "last updated" date on single posts` was enabled. 

This PR makes sure that functionality still works, while also allowing the 'updated date' option.

This came up when digging into #1247

### How to test the changes in this Pull Request:

1. Make sure `Show "last updated" date on single posts` is turned off under Customizer > Template Settings > Post Settings.
2. Update an older post, and publish a brand new post.
3. Inspect the published dates for both; you should just have one `<time>` in the source in both cases:

![image](https://user-images.githubusercontent.com/177561/108241591-db699300-7100-11eb-89ca-b9ff35bc17c1.png)

4. Apply the PR.
5. Confirm that the new post still has one `<time>` in the source, while the updated one now has two:

![image](https://user-images.githubusercontent.com/177561/108241708-f9cf8e80-7100-11eb-9320-66561a00058a.png)

6. Confirm that this is the case wherever the posts appear from the theme -- a single post, archives, search results.
7. Navigate to Customizer > Template Settings > Post Settings and enable  `Show "last updated" date on single posts`.
8. Confirm that your updated date is now displaying on single posts. 
9. Confirm that nothing funky happened to the date in the archives or search, which should not have changed. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

dkoo reviewed (approved):
Sorry I changed the logic back in #1154—didn't realize that both time strings were still needed, even if not shown!

This works as described except:

> Confirm that the new post still has one <time> in the source, while the updated one now has two

Both the brand new post and the updated post had two `time` elements (the brand new one with a modified time that was just a few seconds after the publish time—maybe from an auto-save?), but this didn't affect how things looked or behaved either with or without the `Show "last updated" date on single posts` option enabled.

laurelfulford commented:
> Sorry I changed the logic back in #1154—didn't realize that both time strings were still needed, even if not shown!

No worries! I should've caught that :) 

> Both the brand new post and the updated post had two time elements (the brand new one with a modified time that was just a few seconds after the publish time—maybe from an auto-save?), but this didn't affect how things looked or behaved either with or without the Show "last updated" date on single posts option enabled.

I can recreate this too -- first I tried creating two posts back to back, one really quickly, and the second allowing it to autosave while I messed around. At first both only had one date... but when I went to the archives and back, they both had two -- one example was 14 seconds apart:

```
Published: 2021-02-17T11:01:20-08:00
Modified:  2021-02-17T11:01:34-08:00
```

I double-checked this against the most-recent default theme that handles the updated date this way (Twenty Nineteen), and I get the same result for the same posts. So maybe it's some kind of delay in the full publishing process? 

This PR does restore it to the original behaviour -- and matches Twenty Nineteen -- so I'm going to go ahead and merge it, but if it does turn out to be an issue, we could try switching the `if ( get_the_time( 'U' ) !== get_the_modified_time( 'U' ) )` check to use a different date format that doesn't have seconds :)
